src/main/java/com/onehouse/metadata_extractor/TimelineCommitInstantsUploader.java

```diff
@@ -251,9 +270,13 @@ private CompletableFuture<Checkpoint> uploadInstantsInSequentialBatches(
           Lists.partition(
               filesToUpload, getUploadBatchSize(CommitTimelineType.COMMIT_TIMELINE_TYPE_ARCHIVED));
     } else {
-      batches =
+      Pair<Checkpoint, List<List<File>>> checkpointBatchesPair =
           activeTimelineInstantBatcher.createBatches(
-              filesToUpload, getUploadBatchSize(CommitTimelineType.COMMIT_TIMELINE_TYPE_ACTIVE));
+              filesToUpload,
+              getUploadBatchSize(CommitTimelineType.COMMIT_TIMELINE_TYPE_ACTIVE),
+              checkpoint);
+      batches = checkpointBatchesPair.getRight();
+      checkpoint = checkpointBatchesPair.getLeft();
```

Comment:
can we return just the last incompleteCommitInstant file name? here we are updating the checkpoint obj and in later parts of the code we refer to it as the previous checkpoint which that is not correct (as it has been modified)